src/m365/spfx/commands/project/project-azuredevops-pipeline-add.ts

```diff
@@ -0,0 +1,213 @@
+          }
+
+          const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.siteUrl);
+          if (isValidSharePointUrl !== true) {
+            return isValidSharePointUrl;
+          }
+        }
+
+        if (args.options.loginMethod && SpfxProjectAzureDevOpsPipelineAddCommand.loginMethod.indexOf(args.options.loginMethod) < 0) {
+          return `${args.options.loginMethod} is not a valid login method. Allowed values are ${SpfxProjectAzureDevOpsPipelineAddCommand.loginMethod.join(', ')}`;
+        }
+
+        if (args.options.scope && SpfxProjectAzureDevOpsPipelineAddCommand.scope.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Allowed values are ${SpfxProjectAzureDevOpsPipelineAddCommand.scope.join(', ')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    this.projectRootPath = this.getProjectRoot(process.cwd());
+    if (this.projectRootPath === null) {
+      throw new CommandError(`Couldn't find project root folder`, SpfxProjectAzureDevOpsPipelineAddCommand.ERROR_NO_PROJECT_ROOT_FOLDER);
+    }
+
+    const solutionPackageJsonFile: string = path.join(this.projectRootPath, 'package.json');
+    const packageJson: string = fs.readFileSync(solutionPackageJsonFile, 'utf-8');
+    const solutionName = JSON.parse(packageJson).name;
+
+    if (this.debug) {
+      logger.logToStderr(`Adding Azure DevOps pipeline in the current SPFx project`);
+    }
+
+    try {
+      this.updatePipeline(solutionName, pipeline, args.options);
+      this.savePipeline(pipeline);
+    }
+    catch (error: any) {
+      throw new CommandError(error);
+    }
+  }
+
+  private savePipeline(pipeline: azureDevOpsPipeline): void {
+    const azureDevOpsPath: string = path.join(this.projectRootPath as string, '.azuredevops');
+    fsUtil.ensureDirectory(azureDevOpsPath);
+
+    const pipelinesPath: string = path.join(azureDevOpsPath, 'pipelines');
+    fsUtil.ensureDirectory(pipelinesPath);
+
+    const pipelineFile: string = path.join(pipelinesPath, 'deploy-spfx-solution.yml');
+    fs.writeFileSync(path.resolve(pipelineFile), yaml.stringify(pipeline), 'utf-8');
+  }
+
+  private updatePipeline(solutionName: string, pipeline: azureDevOpsPipeline, options: GlobalOptions): void {
+    if (options.name) {
+      pipeline.name = options.name;
+    }
+    else {
+      delete pipeline.name;
+    }
+
+    if (options.branchName) {
+      pipeline.trigger.branches.include[0] = options.branchName;
+    }
+
+    const version = this.getProjectVersion();
+
+    if (!version) {
+      throw `Unable to determine the version of the current SharePoint Framework project`;
+    }
+
+    const match = version.match(/^(\d+)\.(\d+)\.(\d+)$/);
```

Comment:
Wouldn't it be easier and more robust to parse the version using the semver library which we already have in the project?